pisces/data_sets.py

```diff
@@ -503,7 +503,183 @@ def __init__(self,
         self.input_sampling_hz = float(input_sampling_hz)
         self.spectrogram_preprocessing_config = spectrogram_preprocessing_config
 
-# %% ../nbs/01_data_sets.ipynb 17
+# %% ../nbs/01_data_sets.ipynb 18
+def get_sample_weights(y: np.ndarray) -> np.ndarray:
+     """
+     Calculate sample weights based on the distribution of classes in the data.
+     Doesn't count masked values (-1) in the class distribution.
+     """
+     # Filter out -1 values
+     valid_y = y[y != -1]
+     # Calculate class counts for valid labels only
+     class_counts = np.bincount(valid_y)
+     class_weights = np.where(class_counts > 0, class_counts.sum() / class_counts, 0)
+     # Map valid class weights to corresponding samples in y
+     sample_weights = np.zeros_like(y, dtype=float)
+     for class_index, weight in enumerate(class_weights):
+          sample_weights[y == class_index] = weight
+     # Masked values (-1) in y will have a weight of 0
+     return sample_weights
+
+
+def mask_psg_from_accel(psg: np.ndarray, accel: np.ndarray, 
+                        psg_epoch: int = 30,
+                        accel_sample_rate: float | None = None,
+                        min_epoch_fraction_covered: float = 0.5
+                        ) -> np.ndarray:
+
+    acc_last_index = 0
+    acc_next_index = acc_last_index
+    acc_last_time = accel[acc_last_index, 0]
+    acc_next_time = acc_last_time
+
+    # at least this fraction of 1 epoch must be covered
+    # both in terms of time (no gap longer than 0.5 epochs)
+    # and in terms of expected number of samples in that time.
+    min_epoch_covered = min_epoch_fraction_covered * psg_epoch
+    if accel_sample_rate is None:
+        # median sample step size, if none provided
+        # median to not take into account gaps!
+        accel_sample_rate = np.median(np.diff(accel[:, 0]))
+    min_samples_per = min_epoch_covered / accel_sample_rate
+
+    psg_gap_indices = []
+
+    for (psg_index, psg_sample) in enumerate(psg):
+        epoch_ends = psg_sample[0] + psg_epoch
+
+        # find the last timestamp inside the epoch
+        while (acc_next_time <= epoch_ends and acc_next_index < len(accel)):
+            acc_next_time = accel[acc_next_index, 0]
+            acc_next_index += 1
+        
+        return data_sets
+
+    def find_overlapping_time_section(
```

Comment:
it looks like this function is indented too deeply, it's scope is just `mask_psg_from_accel` as far as I can tell.

Reply:
You're totally right! Nice catch